src/main/java/panda/command/AlterMarkCommand.java

```diff
@@ -0,0 +1,85 @@
+package panda.command;
+import panda.exception.PandaException;
+import panda.exception.OutOfBoundsException;
+import panda.component.*;
+public class AlterMarkCommand extends Command {
+    private int idx;
+    private boolean isMarked;
+
+    /**
+     * Constructs a new AlterMarkCommand.
+     * 
+     * @param idx the index of the task to alter mark.
+     * @param isMarked the logic to mark the task
+     */
+    public AlterMarkCommand(int idx, boolean isMarked) {
+        this.idx = idx;
+        this.isMarked = isMarked;
+    }
+
+    /**
+     * Executes the command on the given TaskList.
+     * 
+     * @param tlist the TaskList on which the command is executed.
+     * @throws PandaException if an error occurs during execution.
+     */
+    public void execute(TaskList tlist) throws PandaException {
+        if(idx - 1 >= tlist.size()) {
+            throw new OutOfBoundsException();
+        }
+        if(isMarked) {
+            tlist.mark(idx);
+        }
+        else {
+            tlist.unmark(idx);
+        }
+    }
+
+    /**
+     * Executes the command on the given TaskList, updates the UI, and saves changes to the cache file.
+     * 
+     * @param tlist the TaskList on which the command is executed.
+     * @param ui the UI to update after execution.
+     * @param cacheFile the cache file to save changes to.
+     * @throws PandaException if an error occurs during execution.
+     */
+    public void execute(TaskList tlist, Ui ui, Storage cacheFile) throws PandaException {
+        if(idx - 1 >= tlist.size()) {
+            throw new OutOfBoundsException();
+        }
+        if(isMarked) {
+            tlist.mark(idx);
+        }
+        else {
```

Comment:
Is the first execute method still needed since the updated code is using the new Ui and Storage classes?